### ISSUE TITLE
SecurityContext not propagated to pods

### ISSUE THREAD

**Issue by Jorricks** [kube-logging/logging-operator#1319]
> Bugs should be filed for issues encountered whilst operating logging-operator.
> You should first attempt to resolve your issues through the community support
> channels, e.g. Slack, in order to rule out individual configuration errors. [#logging-operator](https://slack.banzaicloud.io/)
> Please provide as much detail as possible.

**Describe the bug**:
I am trying to set restricted permissions on the logging operator. As it runs the fluent-bit & config reloader with the root user, it has an elevated risk factor which I am trying to reduce. 
When I set the securityContext in the logging operator, it is not reflecting all the items inside the pod.

**Expected behaviour**:
I expect all the securityContext items to be reflecting on the definition of the fluentd & fluentbit pod.

**Steps to reproduce the bug**:
Create a Logging resource similar to this;
```
apiVersion: logging.banzaicloud.io/v1beta1
kind: Logging
spec:
  fluentd:
    security:
      podSecurityContext:
        fsGroup: 101
        runAsGroup: 101
        runAsNonRoot: true
        runAsUser: 100
        seccompProfile:
          type: RuntimeDefault
      securityContext:
        allowPrivilegeEscalation: false
        capabilities:
          drop:
            - ALL
```
Now inside the fluentd pod, you will see;
```
...
  containers:
  - ..
    securityContext:
      allowPrivilegeEscalation: false
...
  securityContext:
    fsGroup: 101
    runAsGroup: 101
    runAsNonRoot: true
    runAsUser: 100
...
```

The capabilities & seccomp profile are dropped.  For fluent-bit, that makes sense because it needs access to the volume of the host. However, for fluentd I would argue these items should be present on the pod.

**Additional context**:
--

**Environment details**:
- Kubernetes version (e.g. v1.15.2): `v1.24.4+rke2r1`
- Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): RKE (self-hosted)
- logging-operator version (e.g. 2.1.1): 4.0.0
- Install method (e.g. helm or static manifests): HELM
- Logs from the misbehaving component (and any other relevant logs): --
- Resource definition (possibly in YAML format) that caused the issue, without sensitive data: Added above

/kind bug

**Comment by jeremych1000** [comment]
I've run into this as well. I have a policy engine (Kyverno) that blocks any pod that doesn't drop all capabilities. This means the fluentbit daemonset fails to come up for me.

I have 
```
security:
  securityContext:
  allowPrivilegeEscalation: false
  readOnlyRootFilesystem: true
  capabilities:
    drop:
    - "ALL"
   ```

in my logging CR.

**Comment by jeremych1000** [comment]
Also - this does not get passed to the fluentd-configcheck container as well. Both capabilities and seccomp are stripped out.

**Comment by jeremych1000** [comment]
What about the configcheck container as well please @kristofgyuracz ?

**Comment by kristofgyuracz** [comment]
Sorry, that slipped through. See https://github.com/kube-logging/logging-operator/pull/1459

**Comment by kristofgyuracz** [comment]
Merged #1459, please reopen if you encounter this issue again.

**Comment by jeremych1000** [comment]
@kristofgyuracz Doesn't seem to be applied to the node-exporter pod that spins up if you request metrics
